### DISCUSSION TITLE
Relational update is giving unknown 'arg' error.

### DISCUSSION THREAD

**Discussion by sultan-haider** [prisma/prisma#18187] [Q&A]
I have been trying to update a table and a related nested child table as well which has one to one relationship.
<img width="867" alt="Screenshot 2023-03-02 at 6 24 20 PM" src="https://user-images.githubusercontent.com/40316175/222440840-66835343-ee00-467b-9c01-9d7a9bc00229.png">

I am trying to update permissions along with the Workspace members and both 1 to 1 related.
<img width="714" alt="Screenshot 2023-03-02 at 6 12 48 PM" src="https://user-images.githubusercontent.com/40316175/222441079-0e171ddf-8fc7-447a-8cb8-3fb68d0aa559.png">

Now when I am trying to update it I am getting this error
<img width="1566" alt="Screenshot 2023-03-02 at 6 12 54 PM" src="https://user-images.githubusercontent.com/40316175/222441253-a9c1242a-d832-4f7f-b353-bea2b5f5f476.png">

Even though the relation is fine and I am successfully fetching the data.

**Comment by ludralph** [comment]
Hi @sultan-haider 👋 

Can you please share the full schema for Permission as well so I can recreate this and recommend a solution?

**Reply by ludralph** [reply]
Hi @sultan-haider 

it would be really helpful if you could provide the additional information that we asked about last time. If your problem is resolved by now or somehow doesn't feel relevant anymore, you can simply close the Discussion as well.

**Reply by ludralph** [reply]
Hi @sultan-haider, I'll be closing this discussion since there has been no response from you.

**Comment by sebastian-radon** [comment]
I had the same issue ...

In my permission-model I had:

`route        Route?`

In the route-model I had:

`permission             Permission?           @relation(fields: [permission_id], references: [id])`
`permission_id          String?               @unique`

I changed my route-model to:

`permission        Permission?`

And my permission-model to:

`route             Route?                @relation(fields: [route_id], references: [id])`
`route_id          String?               @unique`

I'm still not sure, why it works now. Changing everything back will definitely break the application. I guess it has something to do with the order of the models inside schema.prisma. The permission-model comes first, then the route-model. Maybe someday I'll have time to investigate it further.
